test/viewmodels/CheckYourAnswersViewModelSpec.scala

```diff
@@ -0,0 +1,229 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels
+
+import base.SpecBase
+import helpers.JsonFixtures._
+import models.matching.OrgRegistrationInfo
+import models.register.response.details.AddressResponse
+import models.{Address, Country, ReporterType}
+import org.scalatestplus.play.guice.GuiceOneAppPerSuite
+import pages._
+import play.api.i18n.{Messages, MessagesApi}
+import play.api.mvc.AnyContentAsEmpty
+import play.api.test.FakeRequest
+import uk.gov.hmrc.domain.Nino
+import utils.CountryListFactory
+import viewmodels.checkAnswers.CheckYourAnswersViewModel
+
+import java.time.LocalDate
+
+class CheckYourAnswersViewModelSpec extends SpecBase with GuiceOneAppPerSuite {
+
+  def fakeRequest: FakeRequest[AnyContentAsEmpty.type] = FakeRequest("", "")
+  def messagesApi: MessagesApi                         = app.injector.instanceOf[MessagesApi]
+  implicit def messages: Messages                      = messagesApi.preferred(fakeRequest)
+  val countryListFactory: CountryListFactory           = app.injector.instanceOf[CountryListFactory]
+  val orgRegistrationInfo: OrgRegistrationInfo         = OrgRegistrationInfo(safeId, OrgName, AddressResponse("line1", None, None, None, None, "GB"))
+
+  "CheckYourAnswersViewModel" - {
+
+    "must return required rows for 'business-with-id' flow" in {
+      val userAnswers = emptyUserAnswers
```

Comment:
Wonder if there's a smaller way to do this 😆

Reply:
I'll make it more compact. Thanks 👍

Reply:
not sure if you can - one to look at in some down time

Reply:
I've added a test extension class to provide a `withPage(...)` functionality.